NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -442,7 +448,11 @@ class NeuroID
             }
             dataStore.saveAndClearAllQueuedEvents()
 
-            this.getApplicationContext()?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            this.getApplicationContext()
+                ?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            locationService?.let { ls ->
+                ls.setupLocationCoroutine(getApplicationContext()?.getSystemService(Context.LOCATION_SERVICE) as LocationManager)
+            }
```

Comment:
is there a reason you wouldn't have the locationService.let inside the getApplicationContext you already called?

Reply:
can do